### ISSUE TITLE
DPL-nnn Cardinal - Sample pooling plate

### ISSUE THREAD

**Issue by KatyTaylor** [sanger/limber#759]
**User story**
I need to be able pool the samples on the LCA PBMC plate into wells of a new plate. The samples assigned to the pools, the number of samples to be pooled together per pool and the volume pooled of each sample needs to be determined by the LIMS. 
To do this I need to:

- Create a new plate LCA PBMC pools plate
- Based on the number of samples that are taken forward from the LCA PBMC plate to the LCA PBMC pools plate, the number of pools and the number of samples per pool is determined using the following table (Figure 3). E.g. 96 passed wells on the plate get split into 8 pools of 12 samples per pool. 

![image](https://user-images.githubusercontent.com/10407809/124908166-ef932b80-dfe0-11eb-8a32-ee19dc157c93.png)

_Figure 3: Plexing parameters_

- Samples should be assigned to the pools randomly and not in an ordered fashion. E.g. A1, B1,C1  should not be assigned to pool 1. Wherever possible pools should contain samples from more than one blood collection site. 
- The volume of each sample to be added to the pool needs to be calculated from the previously uploaded Cardinal_PBMC_Count file. 
- 20,000 cells need to be taken from each sample into the pool. The volume that needs to be taken to achieve this needs to be calculated from the live cell count. E.g. 1,000,000 live cells per ml (20,000/1,000,000) = 0.02ml (20ul). 
- A resuspension volume also needs to be specified per pool. The volume is fixed and specified in Figure 4.  

![image](https://user-images.githubusercontent.com/10407809/124908187-f6ba3980-dfe0-11eb-8230-05e1d48b4f34.png)

_Figure 4: resuspension volume_

- The LIMS needs to generate a Hamilton driver file that states: source plate, source well, destination plate, destination well, sample volume and resuspension volume. The destination well identifies the pool e.g. all samples going into well A1 are in a single pool. The sample volume reflects the amount of sample volume that needs to go into the pool (20,000 cells per sample). The resuspension volume is taken from Figure 4 and depends on the number of samples in the pool.

![image](https://user-images.githubusercontent.com/10407809/124908219-fe79de00-dfe0-11eb-94ee-73484c6ae6ec.png)
_Figure 5: Example sample pooling driver file_

**Who are the primary contacts for this story**
Stef L

**Acceptance criteria**
To be considered successful the solution must allow:
- [ ] add a list of acceptance criteria here
- [ ] ...

**Dependencies**
This story is blocked by the following dependencies:
- #<issue_no.>
- sanger/<repo>#<issue_no.>

References
This story has a non-blocking relationship with:
- #<issue_no.>
- sanger/<repo>#<issue_no.>

**Additional context**

**Comment by KatyTaylor** [comment]
Q's:
1. "Based on the number of samples that are taken forward" - as in, ones that weren't failed based on cell count / viability?

**Comment by andrewsparkes** [comment]
See existing models/labware_creators in Limber for custom steps like this e.g. normalised_binned_plate.rb

**Comment by KatyTaylor** [comment]
Collection site will be selected from the 'supplier' dropdown when generating the sample manifests - so is a field on the sample manifest table, which can be accessed through the 'sample_manifest_id' on 'sample'

**Comment by harrietc52** [comment]
Meeting on 14th Sept: Tag Clashes in the PBMC Plate
Attendees: Stef, Harry, Andrew, James, Ben, Steve, Stuart, Eduardo

We discussed how at the creation of the PBMC Pooling plate, it is currently failing due to a tag clash error; On the PBMC Pools Plate, we are adding up to 8 pools, one pool per well, onto the pooling plate. The samples in the pools at this stage are untagged. There is currently tag clash detection in the aliquots table in Sequencescape's database, where a unique index is on tag_1, tag_2 and receptacle. Meaning that multiple samples in one well, with no tags are hitting this constraint. 

We discussed a few options with Stef. And decided to, in summary, add a new column to the aliquots table, which can be added to the unique index, so many untagged samples in one pool are unique. A new story DPL-xxx was created

**Comment by harrietc52** [comment]
![image](https://user-images.githubusercontent.com/10407809/124908166-ef932b80-dfe0-11eb-8a32-ee19dc157c93.png)
